src/Data/Geometry/Geos/Geometry.hs

```diff
@@ -125,8 +127,8 @@ convertLineStringToRaw (LineString cs) s = do
 
 convertPolygonToRaw :: R.Geometry a => Polygon -> SRID -> Geos a
 convertPolygonToRaw (Polygon lrs) s = do
-  ext <- convertLinearRingToRaw (V.head lrs) s
-  inn <- (flip convertLinearRingToRaw $ s) `V.mapM` V.tail lrs
+  ext :: R.GeomConst <- convertLinearRingToRaw (V.head lrs) s
+  inn :: V.Vector R.GeomConst <- (flip convertLinearRingToRaw $ s) `V.mapM` V.tail lrs
   R.createPolygon ext (V.toList inn)  >>= R.setSRID s
 
 convertMultiPointToRaw :: R.Geometry a => MultiPoint -> SRID -> Geos a
```

Comment:
Would I be right in thinking that these other converters also need the GeomConst treatment to stop the sub-objects (Points) of the parent (MultiPoint) trying to free themselves after the parent has already destroyed/free'd their underlying pointers?

Or have I misunderstood (I estimate that at > 50% probability)